Wikipedia/Categories/AFHTTPRequestOperationManager+WMFDesktopRetry.h

```diff
@@ -38,4 +42,19 @@
                                    success:(void (^)(AFHTTPRequestOperation* operation, id responseObject))success
                                    failure:(void (^)(AFHTTPRequestOperation* operation, NSError* error))failure;
 
+
+/**
+ *  Send a @c GET request to the given site, falling back to desktop URL if the mobile URL fails.
+ *
+ *  @param site The site to send the request to, using its API endpoints. First mobile, then desktop.
+ *
+ *  @return A promise which will be resolved with the successful response of the request, or rejected with any errors
+ *          that occur.
+ *
+ *  @see wmf_GETWithMobileURLString:desktopURLString:parameters:retry:success:failure:
+ */
+- (AnyPromise*)wmf_GETWithSite:(MWKSite*)site
```

Comment:
@bgerstle If we pass out an outOperation, we also need to provide a retry block as in the method above. The reason for passing the retry block is because if a retry occurs, then the operation you received initially cannot be used for cancellation.

The other option is not passing back an outOperation if cancellation is not needed (making this more of a convenience method).